internal/git/git.go

```diff
@@ -0,0 +1,65 @@
+package git
+
+import (
+	"fmt"
+	"net/url"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+type Git interface {
```

Comment:
I think we can improve the name by not binding it to `git`, there are other repo types. Also Go-wise, there is a convention to call interface by the action they perform. In this case, maybe `RemoteRepoURLGetter` would be better?